lib/components/formik/BlockNavigation.js

```diff
@@ -3,10 +3,10 @@ import Modal from "../Modal";
 import { useFormikContext } from "formik";
 import { useNavPrompt } from "react-router-nav-prompt";
 
-export default function NavigationHandler() {
+export default function NavigationHandler({ isDirty = false }) {
   const { dirty } = useFormikContext();
   const { blocked, hidePrompt, navigate } = useNavPrompt({
-    shouldBlock: Boolean(dirty)
+    shouldBlock: Boolean(dirty) || isDirty,
```

Comment:
@vyshnavrajs2019 _a
```suggestion
    shouldBlock: isDirty || Boolean(dirty),
```
Otherwise, how will it work in neetoChat?

Reply:
@edwinbbu They are the same. If the formik state itself is dirty we can use that value (which obviously is true). If. that's not the case shouldBlock depends on the isDirty prop.

Reply:
@edwinbbu _A